README.md

```diff
@@ -104,41 +118,49 @@ val fetchThree: Fetch[(String, String, String)] = (fetchString(1) |@| fetchStrin
 When executing the above fetch, note how the three identities get batched and the data source is only queried once.
 
 ```scala
-val result: (String, String, String) = fetchThree.runA[Eval].value
-// ToStringSource OneAnd(1,List(2, 3))
-// result: (String, String, String) = (1,2,3)
+fetchThree.runA[Id]
+// [42] Many ToString OneAnd(1,List(2, 3))
+// res5: cats.Id[(String, String, String)] = (1,2,3)
 ```
 
-## Concurrency
+## Parallelism
 
-If we combine two independent fetches from different data sources, the fetches will be run concurrently. First, let's add a data source that fetches a string's size.
+If we combine two independent fetches from different data sources, the fetches can be run in parallel. First, let's add a data source that fetches a string's size.
+
+This time, instead of creating the results with `Query#later` we are going to do it with `Query#async` for emulating an asynchronous data source.
 
 ```scala
 implicit object LengthSource extends DataSource[String, Int]{
-  override def fetch(ids: NonEmptyList[String]): Eval[Map[String, Int]] = {
-    Eval.later({
-      println(s"LengthSource $ids")
-      ids.unwrap.map(i => (i, i.size)).toMap
+  override def fetchOne(id: String): Query[Option[Int]] = {
+    Query.async((ok, fail) => {
+      println(s"[${Thread.currentThread.getId}] One Length $id")
+      ok(Option(id.size))
+    })
+  }
+  override def fetchMany(ids: NonEmptyList[String]): Query[Map[String, Int]] = {
+    Query.async((ok, fail) => {
+      println(s"[${Thread.currentThread.getId}] Many Length $ids")
+      ok(ids.unwrap.map(i => (i, i.size)).toMap)
     })
   }
 }
 
 def fetchLength(s: String): Fetch[Int] = Fetch(s)
 ```
 
-And now we can easily recieve data from the two sources in a single fetch. 
+And now we can easily receive data from the two sources in a single fetch. 
```

Comment:
Does this have to do with the face that we've marked one as being explicitly async? What if they were both `Query.later`? Or both used `Query.async`? Perhaps instead of "And now..." it could be "Because we've marked one of these data sources as being asynchronous with `Query.later`, Fetch knows that it can execute independent fetches". Or something like that. Could also be reworked into the next paragraph.